### ISSUE TITLE
Style/WordArray with nested arrays of which some are word arrays

### ISSUE THREAD

**Issue by dziemba** [bbatsov/rubocop#2767]
Given this code:
```ruby
p [
  ['a', 'b', 'c'],
  ['d e', 'f']
]
```

rubocop currently (0.36.0) reports this notice:

````
x.rb:2:3: C: Style/WordArray: Use %w or %W for an array of words.
  ['a', 'b', 'c'],
  ^^^^^^^^^^^^^^^
```

and autocorrects to:

```ruby
p [
  %w(a b c),
  ['d e', 'f']
]
```

In my opinion the mix of `%w()` and `[]` looks weird and should be avoided (it should not report the first line of `[]` as an offense). 

I am not sure what the exact criteria should be, but it should at least work for nested arrays. 

What do you think?

**Comment by alexdowad** [comment]
I think the cop is OK as is. We could come up with more and more complicated rules, but that is an endless road which leads somewhere we don't want to go.

Almost every cop has the same problem. A machine is incapable of understanding everything about the code which a human can. There are always cases where the machine makes a questionable call. If what it is telling you to do is really bad, you can disable the cop globally or locally (using a `# rubocop:disable` comment).

It can be beneficial to add a new rule if:

1) The rule is simple and easy to describe, and
2) It consistently leads to better results, and
3) The situation it addresses is common enough to be worth the trouble

Number 2 is a tricky one. Unless you have thought things out very, very thoroughly, a new rule which gives better results in one situation will often produce worse results in another.

**Comment by alexdowad** [comment]
Can we close this one?

**Comment by bbatsov** [comment]
We sure can.

**Comment by phantomwhale** [comment]
Bumped into this with an array of tuples - can understand why the decision was made not to implement it.

Ended up adding `MinSize: 3` to my `Style/WordArray` config, which did the job in my case
